gradle/render-javadoc.gradle

```diff
@@ -294,35 +305,41 @@ configure(project(':solr:core')) {
   plugins.withType(JavaPlugin) {
     // specialized to ONLY depend on solrj
     renderJavadoc {
-      dependsOn ':solr:solrj:renderJavadoc'
-      linkHref += [ '../solr-solrj' ]
+      [':solr:solrj'].collect { path ->
+        dependsOn "${path}:renderJavadoc"
+        linkHref += [ "../" + pathToDocdir(path) ]
+      }
     }
   }
 }
 
 configure(subprojects.findAll { it.path.startsWith(':solr:contrib') }) {
   plugins.withType(JavaPlugin) {
     renderJavadoc {
-      dependsOn ':solr:solrj:renderJavadoc'
-      dependsOn ':solr:core:renderJavadoc'
-      linkHref += [ '../solr-solrj', '../solr-core' ]
+      [':solr:solrj', ':solr:core'].collect { path ->
+        dependsOn "${path}:renderJavadoc"
+        linkHref += [ "../../" + pathToDocdir(path) ]
+      }
     }
   }
 }
 
 configure(project(':solr:contrib:dataimporthandler-extras')) {
   plugins.withType(JavaPlugin) {
     renderJavadoc {
-      dependsOn ':solr:contrib:dataimporthandler:renderJavadoc'
-      linkHref += [ '../solr-dataimporthandler' ]
+      [':solr:contrib:dataimporthandler'].collect { path ->
+        dependsOn "${path}:renderJavadoc"
+        linkHref += [ "../../" + pathToDocdir(path) ]
+      }
     }
   }
 }
 
+
 configure(project(':solr:contrib:extraction')) {
   plugins.withType(JavaPlugin) {
     ext {
-      javadocDestDir = "${javadocRoot}/solr-cell"
+      javadocDestDir = "${javadocRoot}/contrib/cell"
```

Comment:
I'm not sure this tweak is needed for gradle build. Simply, "solr/contrib/extraction" (same as other contrib modules) may be okay?

Reply:
The "solr-cell" is what it's called in ant. I'm not sure why. I would prefer to stick to folder/ project path naming here too (so contrib/extraction for me).

Reply:
I removed the entire configure block, so now the docs will go to `contrib/extraction`.